User Guide Update - Annotations button (#8494)
geosolutions-it/MapStore2: docs/user-guide/annotations.md

# Annotations 

> ********************

Reviewer: lint error here, please fix it

[Mapstore](https://mapstore.geosolutionsgroup.com/mapstore/#/) lets you enrich the map with special features which expose additional information, mark particular position on the map and so on.
Those features make up the so called **Annotations** layers.

> Starting from a new map or an already existing one, the editor can access the **Annotations** <img src="../img/button/annotations2.jpg" class="ms-docbutton" style="max-height:30px;"/> button from the [TOC](toc.md) <img src="../img/button/show-layers.jpg" class="ms-docbutton" style="max-height:30px;"/> panel on the top-left corner of the map viewer. The annotation panel will open:

Reviewer: please add here an image of the toolbar
like
![image](https://user-images.githubusercontent.com/11991428/191512574-ade22142-df22-4c0e-8736-e92dc3c1d6e6.png)

> <img src="../img/annotations/annotation_tool.jpg" class="ms-docimage" style="max-width:500px;" />

Reviewer: no need for this to be so long in height
![image](https://user-images.githubusercontent.com/11991428/191512706-b76544b2-070c-4fd3-a320-f396cfde590c.png)

<img src="../img/annotations/add_vertex_button.gif" class="ms-docimage" style="max-width:700px;"/>

!!! note
    If the vertices are invalid, they are notified with a red exclamation point.
    <img src="../img/annotations/invalid_vertex.jpg" class="ms-docimage" style="max-width:500px;" />
    In this case, it is not possible to add new geometry or save the annotation until a valid value is entered. It is still possible to interact with the geometries already present in the annotation, by zooming in on it or deleting it, as follows:
    <img src="../img/annotations/invalid_vertex2.gif" class="ms-docimage" style="max-width:500px;" />

* Customize the **Style** of the annotation, as explained in the following paragraph.

Once the geometry has been saved through the **Save** <img src="../img/button/save_button.jpg" class="ms-docbutton" /> button, for each geometry created, the editor can perform the following operations:

* **Zoom** to the geometry annotation on map through the <img src="../img/button/zoom-feature.jpg" class="ms-docbutton" /> button

* **Delete** the geometry annotation through the <img src="../img/button/delete_white_button.jpg" class="ms-docbutton" /> button

> Once all the *Geometries* have been created, the user can save the annotation through the **Save** <img src="../img/button/save_button.jpg" class="ms-docbutton" /> button and it will be visible:

Reviewer: ```suggestion
Once all the *Geometries* have been created, the user can save the annotation through the **Save** <img src="../img/button/save_button.jpg" class="ms-docbutton" /> button that will be visible in the annotation list:
```

> * in the annotation list

Reviewer: ```suggestion

```

<img src="../img/annotations/annotation1.jpg" class="ms-docimage" />

> * once the annotations panel is closed, trought the <img src="../img/button/x2.jpg" class="ms-docbutton" /> button, in the [TOC](toc.md) as annotations layer

Reviewer: ```suggestion
Then, if not present, a new **Annotations**  layer will be created and added to the [TOC](toc.md) <img src="../img/button/x2.jpg" class="ms-docbutton" /> 
```

Managing Annotations
--------------------

> Once annotations are added to the [TOC](toc.md), the user can **Manage** it by clicking to <img src="../img/button/edit_button.jpg" class="ms-docbutton" /> button from the TOC toolbar and the *Main Annotations panel* will be open.

Reviewer: ```suggestion
Once annotations are added to the [TOC](toc.md), the user can **Manage** them by clicking to <img src="../img/button/edit_button.jpg" class="ms-docbutton" /> button from the TOC toolbar and the *Main Annotations panel* will be open.
```

> Once annotations are added to the [TOC](toc.md), the user can **Manage** it by clicking to <img src="../img/button/edit_button.jpg" class="ms-docbutton" /> button from the TOC toolbar and the *Main Annotations panel* will be open.

Reviewer: maybe we can create a new section "Main Annotations panel" similar to Managing Annotations that can be linked here?

<img src="../img/annotations/annotations_main_panel.jpg" class="ms-docimage" style="max-width:500px;" />

> From it, the editor is allowed to:

Reviewer: here you use the word editor and in other places the word user, can you align these to editor or user?

* **Download** a file with all the existing annotations by clicking on <img src="../img/button/download_annotation_button.jpg" class="ms-docbutton" > button

* **Upload** annotations from a valid `json` file by clicking on <img src="../img/button/upload_annotation_button.jpg" class="ms-docbutton" > button

* **Zoom** an annotation on map by clicking on <img src="../img/button/white-zoom.jpg" class="ms-docbutton" /> button

* **Show/Hide** an annotation on the map by clicking on <img src="../img/button/eyeon.jpg" class="ms-docbutton" /> button

From the *Main Annotations Panel*, by selecting an annotation from the list, the user is returned to the *Annotation Viewer* where the annotation can be edited.

<img src="../img/annotations/annotation_toolbar.jpg" class="ms-docimage" style="max-width:500px;"/>

> In particular, the editor can:

Reviewer: isn't this part a bit mixed to Styling annotations, maybe you can move the gif here to that section
